File: 백준/Silver/1449. 수리공 항승/수리공 항승.py




N ,L = map(int, input().split())
nums =  list( map(int,input().split())) 
nums.sort() 

cnt = 1 
pre = nums[0]+ L  -0.5  
for n in nums[1:]:
    
    if pre < n:
        cnt +=1 
        pre = n+L-0.5 

print(cnt) 

File: 백준/Silver/1449. 수리공 항승/수리공 항승.cc
#include <iostream>
#include <vector> 
#include <string> 
#include <algorithm> 
using namespace std; 



int main() { 

    int N , L ; 
    cin >> N >> L ; 
    
    int n ; 
    vector<int> nums ;
    for(int  i =0; i < N ; i++) { 
        cin >> n;   
        nums.push_back(n ) ; } 

    sort(nums.begin() , nums.end() ) ;

    int cnt=1; 
    int pre = nums[0] +L - 0.5 ; 

    for( int i = 1 ; i < nums.size() ; i++){ 
        if ( pre < nums[i] ) { 
            cnt ++ ; 
            pre = nums[i] + L - 0.5 ; } } 

    cout << cnt ; 




} 
